crates/kubelet/src/pod/mod.rs

```diff
@@ -181,19 +183,37 @@ impl Pod {
             Phase::Running
         };
 
-        let json_status = serde_json::json!(
+        // TODO: init container statuses need to be reported through initContainerStatuses
+
+        let mut json_status = serde_json::json!(
             {
                 "metadata": {
                     "resourceVersion": "",
                 },
                 "status": {
                     "phase": phase,
-                    "message": status.message,
```

Comment:
Is it not possible to insert the correct json value here directly?

Reply:
@rylev It's easy if the value is "clear the setting" (`None`) or "set it to X" (`Some(X)`) but I couldn't see how to specify "don't include the element at all" which is what k8s seems to need to allow it to be left unchanged.

Actually... I guess we are retrieving the existing pod status, so for leave unchanged I could read the existing status and write it back explicitly.  There's a race condition with that but maybe it's not an issue.